prediction/hostedmodels.js

```diff
@@ -0,0 +1,62 @@
+// Copyright 2016, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// [START predict]
+var google = require('googleapis');
+var hostedmodels = google.prediction('v1.6').hostedmodels;
+
+function predict(callback) {
+  
+  google.auth.getApplicationDefault(function(err, authClient) {
+    if (err) {
+      return callback(err);
+    }
+    // The createScopedRequired method returns true when running on GAE or a
+    // local developer machine. In that case, the desired scopes must be passed
+    // in manually. When the code is  running in GCE or a Managed VM, the scopes
+    // are pulled from the GCE metadata server.
+    // See https://cloud.google.com/compute/docs/authentication for more
+    // information. 
+    if (authClient.createScopedRequired && authClient.createScopedRequired()) {
```

Comment:
This really shouldn't be required, if it is, we should open a bug on the API client. Python's does this automatically.